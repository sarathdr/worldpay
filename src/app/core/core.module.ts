import { APP_INITIALIZER, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ConfigService } from './services/config.service';
import { PaymentService } from './services/payment.service';
import { CartService } from './services/cart.service';
import { HttpClientModule } from '@angular/common/http';
import { FormsModule } from '@angular/forms';


export const fetchConfig = (appConfig: ConfigService) => {
  return () => appConfig.getPaymentUrls();
};

@NgModule({
  imports: [
    CommonModule,
    FormsModule
  ],
  providers: [
    ConfigService,
    PaymentService,
    HttpClientModule,
    CartService,
    {
      provide: APP_INITIALIZER,
      useFactory: fetchConfig,
      deps: [ConfigService],
      multi: true
    }
  ]
})
export class CoreModule { }
