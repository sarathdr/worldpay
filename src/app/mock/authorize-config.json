{
  "_links": {
    "resourceTree": {
      "href": "https://try.access.worldpay.com/payments/resourceTree"
    },
    "payments:authorize": {
      "href": "https://try.access.worldpay.com/payments/authorizations"
    },
    "payments:migrateRecurringAuthorize": {
      "href": "https://try.access.worldpay.com/payments/authorizations/migrateRecurring"
    },
    "payments:migrateCardOnFileAuthorize": {
      "href": "https://try.access.worldpay.com/payments/authorizations/migrateCardOnFile"
    },
    "payments:events": {
      "href": "https://try.access.worldpay.com/payments/events"
    },
    "payments:recurringMandate": {
      "href": "https://try.access.worldpay.com/payments/recurringMandates"
    },
    "curies": [
      {
        "name": "payments",
        "href": "https://try.access.worldpay.com/rels/payments/{rel}",
        "templated": true
      }
    ]
  }
}
