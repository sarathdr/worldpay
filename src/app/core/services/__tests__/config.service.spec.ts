import { TestBed } from '@angular/core/testing';
import { HttpClient } from '@angular/common/http';
import { cold } from 'jasmine-marbles';
import { ConfigService } from '../config.service';
import * as authorize from '../../../mock/authorize-config.json';

describe('ConfigService', () => {

  let service: ConfigService;
  let get: jest.Mock;

  beforeEach(() => {
    get = jest.fn();

    TestBed.configureTestingModule({
      providers: [
        ConfigService,
        {
          provide: HttpClient,
          useValue: {
            get
          },
        }
      ]
    });

    service = TestBed.get(ConfigService);

  });

  describe('getPaymentUrls ', () => {

    it('should get the url configs', () => {
      const response$ = cold('-a|', { a: authorize });
      get.mockReturnValue(response$);

      service.getPaymentUrls().then((item) => {
        expect(item).toEqual(authorize);
        expect(service.getAuthorizePaymentUrl()).toMatchSnapshot();
      });
    });

    it('should handle error', () => {
      const response$ = cold('-#', {}, { status: 400 });
      get.mockReturnValue(response$);

      service.getPaymentUrls().catch(errors => {
        expect(errors).toMatchSnapshot();
      });
    });

  });

});
