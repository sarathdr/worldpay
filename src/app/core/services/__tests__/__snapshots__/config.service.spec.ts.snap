// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigService getPaymentUrls  should get the url configs 1`] = `"https://try.access.worldpay.com/payments/authorizations"`;

exports[`ConfigService getPaymentUrls  should handle error 1`] = `
Object {
  "status": 400,
}
`;
